lib/amazing_print/formatters/base_formatter.rb

```diff
@@ -141,6 +141,33 @@ def align(value, width)
           value
         end
       end
+
+      #
+      # Depth related methods
+      #-----------------------------------------
+      def depth
+        inspector.current_depth
+      end
+
+      def depth_limit
+        inspector.max_depth
+      end
+
+      def track_depth(&blk)
+        inspector.increase_depth(&blk)
+      end
+
+      def reached_depth_limit
+        !depth_limit.nil? && depth >= depth_limit
```

Comment:
Would be nice for this to live in something like `DepthTracker#reached_depth_limit?`